File: Frontend/src/services/QuestionService.js
import axios from 'axios';
import {base_url} from './base_url';
import { JwtToken } from './JwtToken';

class QuestionService{

    create(question){
        return axios.post(`${base_url}/quiz-service/question/`,question,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

    get(qid){
        return axios.get(`${base_url}/quiz-service/question/quiz/${qid}`,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

    getSIngleQuestion(quesid){
        return axios.get(`${base_url}/quiz-service/question/${quesid}`,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

    delete(quesid){
        return axios.delete(`${base_url}/quiz-service/question/${quesid}`,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }
    update(question){
        return axios.put(`${base_url}/quiz-service/question/`,question,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

}

export default new QuestionService();

File: Backend/quiz/src/main/java/com/quiz/service/QuestionService.java
package com.quiz.service;

import java.util.*;

import com.quiz.model.Question;
import com.quiz.model.Quiz;

public interface QuestionService {

	public Question addQuestion(Question question);
	
	public Question updateQuestion(Question question);
	
	public Set<Question> getAllQuestions();
	
	public Question getSingleQuestion(Long questionId);
	
	public Set<Question> getQuestionOfQuiz(Quiz quiz);
	
	public void deleteQuestion(Long quesId);
}
